sdk/core/src/response.rs

```diff
@@ -108,15 +88,66 @@ impl CollectedResponse {
         &self.body
     }
 
-    /// Into the body
-    pub fn into_body(self) -> Bytes {
-        self.body
-    }
-
     /// From a response
     pub async fn from_response(response: Response) -> crate::Result<Self> {
         let (status, headers, body) = response.deconstruct();
-        let body = collect_pinned_stream(body).await?;
+        let body = body.collect().await?;
         Ok(Self::new(status, headers, body))
     }
 }
+
+/// A response body stream
+///
+/// This body can either be streamed or collected into `Bytes`
+#[pin_project::pin_project]
+pub struct ResponseBody(#[pin] PinnedStream);
+
+impl ResponseBody {
+    fn new(stream: PinnedStream) -> Self {
+        Self(stream)
+    }
+
+    /// Collect the stream into a `Bytes` collection
+    pub async fn collect(self) -> crate::Result<Bytes> {
+        collect_pinned_stream(self.0).await
```

Comment:
Can we inline `collect_pinned_stream` here? I don't believe it's used anywhere else.